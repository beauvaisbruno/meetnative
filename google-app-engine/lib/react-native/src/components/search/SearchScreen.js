"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const PageTemplate_1 = __importDefault(require("../PageTemplate"));
const react_redux_1 = require("react-redux");
const lang_1 = __importDefault(require("./lang"));
const reducerHelper_1 = require("../../utils/reducerHelper");
const searchReducer_1 = require("./searchReducer");
const react_navigation_1 = require("react-navigation");
const Avatars_1 = require("../profile/img/Avatars");
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
const react_native_auto_height_image_1 = __importDefault(
  require("react-native-auto-height-image")
);
const utils_1 = require("../../utils/utils");
const utilsUi_1 = require("../../utils/utilsUi");
const custom_1 = __importDefault(
  require("../../config/native-base-theme/variables/custom")
);
const Loading_1 = __importDefault(require("../../utils/Loading"));
class SearchScreen extends React.Component {
  componentWillUnmount() {
    this.navListener.remove();
  }
  componentDidMount() {
    this.navListener = this.props.navigation.addListener(
      "willFocus",
      payload => {
        this.props.doSearchProfile();
      }
    );
  }
  render() {
    const { fetching, error, data } = this.props.searchProfiles;
    const {
      lat: userLat,
      lon: userLon,
      langsToTeach: userTeach,
      langsToLearn: userLearn
    } = this.props.userProfile;
    const rows = [];
    if (data) {
      const s = utilsUi_1.font(10);
      const m = utilsUi_1.font(15);
      const l = utilsUi_1.font(20);
      data.forEach(profile => {
        rows.push(
          React.createElement(
            native_base_1.View,
            { key: profile.id, style: { flexDirection: "row" } },
            React.createElement(
              native_base_1.View,
              {
                key: "avatar",
                style: {
                  position: "relative",
                  width: react_native_responsive_screen_1.widthPercentageToDP(
                    "50%"
                  )
                }
              },
              React.createElement(react_native_auto_height_image_1.default, {
                source: Avatars_1.avatars[profile.avatar],
                width: react_native_responsive_screen_1.widthPercentageToDP(
                  "50%"
                )
              }),
              React.createElement(
                native_base_1.Button,
                {
                  block: true,
                  rounded: true,
                  success: true,
                  onPress: () =>
                    this.props.navigation.navigate("RoomScreen", {
                      contactProfile: profile
                    }),
                  testID: "contact-" + profile.id,
                  style: {
                    backgroundColor: "rgba(255, 153, 0, 0.7)",
                    position: "absolute",
                    bottom: 10,
                    alignSelf: "center"
                  }
                },
                React.createElement(
                  native_base_1.Text,
                  null,
                  lang_1.default("Contact")
                )
              )
            ),
            React.createElement(
              native_base_1.View,
              {
                key: "description",
                style: {
                  width: react_native_responsive_screen_1.widthPercentageToDP(
                    "50%"
                  ),
                  paddingLeft: 5
                }
              },
              React.createElement(
                native_base_1.View,
                { style: { flexDirection: "row" } },
                React.createElement(
                  native_base_1.Text,
                  { style: { fontWeight: "bold", fontSize: l } },
                  profile.pseudo
                ),
                React.createElement(native_base_1.Icon, {
                  name: profile.isMale ? "gender-male" : "gender-female",
                  type: "MaterialCommunityIcons",
                  style: {
                    color: profile.isMale
                      ? custom_1.default.brandPrimary
                      : custom_1.default.brandSecondary
                  },
                  fontSize: l
                })
              ),
              React.createElement(
                native_base_1.Text,
                {
                  style: { color: custom_1.default.brandPrimary, fontSize: m }
                },
                profile.city,
                " ",
                utils_1.distanceKm(userLat, userLon, profile.lat, profile.lon),
                "km"
              ),
              React.createElement(
                native_base_1.Text,
                {
                  style: { color: custom_1.default.textGreyLight, fontSize: s }
                },
                profile.profession,
                " - ",
                profile.age + lang_1.default("yo")
              ),
              React.createElement(
                native_base_1.Text,
                { style: { fontWeight: "bold", fontSize: s } },
                lang_1.default("Speak")
              ),
              utilsUi_1.teachRows(profile, s, userLearn),
              React.createElement(
                native_base_1.Text,
                { style: { fontWeight: "bold", fontSize: s } },
                lang_1.default("Learn")
              ),
              utilsUi_1.learnRows(profile, s, userTeach)
            )
          )
        );
      });
    }
    return React.createElement(
      native_base_1.View,
      null,
      fetching && React.createElement(Loading_1.default, null),
      error &&
        React.createElement(
          native_base_1.Text,
          { style: { color: custom_1.default.brandDanger } },
          lang_1.default("Impossible to load profiles")
        ),
      rows.length > 0 && (fetching || rows),
      rows.length === 0 &&
        (fetching ||
          React.createElement(
            native_base_1.View,
            null,
            React.createElement(
              native_base_1.Text,
              { style: { fontStyle: "italic", textAlign: "center" } },
              lang_1.default("No user match your profile")
            )
          ))
    );
  }
}
exports.SearchScreen = SearchScreen;
exports.mapStateToProps = state =>
  Object.assign({}, state.search, {
    userProfile: state.profile.profileLoad.data
  });
exports.mapDispatchToProps = dispatch => {
  return {
    doSearchProfile: () =>
      dispatch(
        reducerHelper_1.fetchingAction(
          searchReducer_1.SearchActions.searchProfiles
        )
      )
  };
};
exports.default = react_redux_1.connect(
  exports.mapStateToProps,
  exports.mapDispatchToProps
)(
  PageTemplate_1.default(
    react_navigation_1.withNavigation(SearchScreen),
    lang_1.default("Find a partner"),
    false
  )
);
//# sourceMappingURL=SearchScreen.js.map
