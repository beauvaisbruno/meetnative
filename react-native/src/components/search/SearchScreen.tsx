import * as React from "react";
import { Button, Icon, Spinner, Text, View } from "native-base";
import PageTemplate from "../PageTemplate";
import { connect } from "react-redux";
import lang from "./lang";
import { IApplicationState, IProfile } from "../../rootReducers";
import { Dispatch } from "redux";
import { fetchingAction } from "../../utils/reducerHelper";
import { ISearchState, SearchActions } from "./searchReducer";
import {
  NavigationEventSubscription,
  NavigationInjectedProps,
  withNavigation
} from "react-navigation";
import { avatars } from "../profile/img/Avatars";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import AutoHeightImage from "react-native-auto-height-image";
import { distanceKm } from "../../utils/utils";
import { font, learnRows, teachRows } from "../../utils/utilsUi";
import custom from "../../config/native-base-theme/variables/custom";
import Loading from "../../utils/Loading";

export type IProps = ISearchState &
  NavigationInjectedProps & {
    doSearchProfile: () => void;
    userProfile: IProfile;
  };

export class SearchScreen extends React.Component<IProps> {
  navListener: NavigationEventSubscription | undefined;
  public componentWillUnmount() {
    if (this.navListener)
      this.navListener.remove();
  }
  public componentDidMount() {
    this.navListener = this.props.navigation.addListener(
      "willFocus",
      payload => {
        this.props.doSearchProfile();
      }
    );
  }

  render() {
    const { fetching, error, data } = this.props.searchProfiles;
    const {
      lat: userLat,
      lon: userLon,
      langsToTeach: userTeach,
      langsToLearn: userLearn
    } = this.props.userProfile;

    const rows: JSX.Element[] = [];
    if (data) {
      const s = font(10);
      const m = font(15);
      const l = font(20);
      data.forEach((profile: IProfile) => {
        rows.push(
          <View key={profile.id} style={{ flexDirection: "row" }}>
            <View
              key={"avatar"}
              style={{ position: "relative", width: wp("50%") }}
            >
              <AutoHeightImage
                source={avatars[profile.avatar] as number}
                width={wp("50%")}
              />
              <Button
                block
                rounded
                success
                onPress={() =>
                  this.props.navigation.navigate("RoomScreen", {
                    contactProfile: profile
                  })
                }
                testID={"contact-" + profile.id}
                style={{
                  backgroundColor: "rgba(255, 153, 0, 0.7)",
                  position: "absolute",
                  bottom: 10,
                  alignSelf: "center"
                }}
              >
                <Text>{lang("Contact")}</Text>
              </Button>
            </View>
            <View
              key={"description"}
              style={{ width: wp("50%"), paddingLeft: 5 }}
            >
              <View style={{ flexDirection: "row" }}>
                <Text style={{ fontWeight: "bold", fontSize: l }}>
                  {profile.pseudo}
                </Text>
                <Icon
                  name={profile.isMale ? "gender-male" : "gender-female"}
                  type="MaterialCommunityIcons"
                  style={{
                    color: profile.isMale
                      ? custom.brandPrimary
                      : custom.brandSecondary
                  }}
                  fontSize={l}
                />
              </View>
              <Text style={{ color: custom.brandPrimary, fontSize: m }}>
                {profile.city}{" "}
                {distanceKm(userLat!, userLon!, profile.lat!, profile.lon!)}km
              </Text>
              <Text style={{ color: custom.textGreyLight, fontSize: s }}>
                {profile.profession}
                {" - "}
                {profile.age + lang("yo")}
              </Text>
              <Text style={{ fontWeight: "bold", fontSize: s }}>
                {lang("Speak")}
              </Text>
              {teachRows(profile, s, userLearn)}
              <Text style={{ fontWeight: "bold", fontSize: s }}>
                {lang("Learn")}
              </Text>
              {learnRows(profile, s, userTeach)}
            </View>
          </View>
        );
      });
    }
    return (
      <View>
        {fetching && <Loading />}
        {error && (
          <Text style={{ color: custom.brandDanger }}>
            {lang("Impossible to load profiles")}
          </Text>
        )}
        {rows.length > 0 && (fetching || rows)}
        {rows.length === 0 &&
          (fetching || (
            <View>
              <Text style={{ fontStyle: "italic", textAlign: "center" }}>
                {lang("No user match your profile")}
              </Text>
            </View>
          ))}
      </View>
    );
  }
}

export const mapStateToProps = (state: IApplicationState) => ({
  ...state.search,
  userProfile: state.profile.profileLoad.data
});

export const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    doSearchProfile: () =>
      dispatch(fetchingAction(SearchActions.searchProfiles))
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PageTemplate(withNavigation(SearchScreen), lang("Find a partner"), false));
