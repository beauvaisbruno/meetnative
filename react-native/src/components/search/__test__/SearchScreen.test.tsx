import { shallow } from "enzyme";
import * as React from "react";
import { SearchScreen } from "../SearchScreen";
import toJson from "enzyme-to-json";
const doSearchProfile = jest.fn();
const navigate = jest.fn();
const addListener = jest.fn();

const contactProfile = {
  langsToLearn: {
    english: "beginner",
    french: "fluent",
    german: "intermediate"
  },
  profession: "Legal Assistant",
  city: "Genève",
  geohash: 872962206,
  langsToTeach: {
    french: "fluent",
    italian: "bilingual",
    spanish: "native"
  },
  lon: 6.107057129141845,
  pseudo: "Maëlle",
  avatar: "pelican",
  lat: 46.08995392215012,
  age: 23,
  id: "IEY3I5fpxQilwKzPt0Sc"
};
const initialProps = {
  doSearchProfile,
  navigation: { navigate, addListener },
  userProfile: {
    langsToLearn: {
      french: "intermediate",
      italian: "beginner",
      spanish: "fluent"
    },
    age: 23,
    lat: 46.144516,
    pseudo: "pseudo",
    lon: 6.081338,
    langsToTeach: {
      english: "native",
      spanish: "fluent"
    },
    geohash: 3661469831164302,
    city: "Genève",
    avatar: "cute",
    profession: "profession"
  },
  searchProfiles: {
    fetching: false,
    error: false,
    data: [contactProfile]
  }
};

it("SearchScreen", () => {
  let tree = shallow(
    // @ts-ignore
    <SearchScreen {...initialProps} />
  );
  expect(toJson(tree)).toMatchSnapshot();
  addListener.mock.calls[0][1]();
  expect(doSearchProfile).toBeCalled();

  tree
    .find("Styled(Button)")
    .props()
    // @ts-ignore
    .onPress();
  expect(navigate).toBeCalledWith("RoomScreen", {
    contactProfile
  });
  tree = shallow(
    // @ts-ignore
    <SearchScreen
      {...initialProps}
      searchProfiles={{
        fetching: true,
        error: false,
        data: undefined
      }}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
  tree = shallow(
    // @ts-ignore
    <SearchScreen
      {...initialProps}
      searchProfiles={{
        fetching: false,
        error: true,
        data: undefined
      }}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
});
