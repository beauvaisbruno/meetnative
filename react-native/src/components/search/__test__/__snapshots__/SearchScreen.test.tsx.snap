// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    key="IEY3I5fpxQilwKzPt0Sc"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Styled(ViewNB)
      key="avatar"
      style={
        Object {
          "position": "relative",
          "width": 375,
        }
      }
    >
      <ErrorableImage
        source={
          Object {
            "testUri": "../../../src/components/profile/img/pelican.png",
          }
        }
        width={375}
      />
      <Styled(Button)
        block={true}
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgba(255, 153, 0, 0.7)",
            "bottom": 10,
            "position": "absolute",
          }
        }
        success={true}
        testID="contact-IEY3I5fpxQilwKzPt0Sc"
      >
        <Styled(Text)>
          Contact
        </Styled(Text)>
      </Styled(Button)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      key="description"
      style={
        Object {
          "paddingLeft": 5,
          "width": 375,
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "fontSize": 47,
              "fontWeight": "bold",
            }
          }
        >
          Maëlle
        </Styled(Text)>
        <Styled(Icon)
          fontSize={47}
          name="gender-female"
          style={
            Object {
              "color": "#ff9900",
            }
          }
          type="MaterialCommunityIcons"
        />
      </Styled(ViewNB)>
      <Styled(Text)
        style={
          Object {
            "color": "#8e7cc3",
            "fontSize": 35,
          }
        }
      >
        Genève
         
        6
        km
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "color": "#919191",
            "fontSize": 24,
          }
        }
      >
        Legal Assistant
         - 
        23yo
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Speak
      </Styled(Text)>
      <Styled(Text)
        key="teach-french"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        French Fluent
      </Styled(Text)>
      <Styled(Text)
        key="teach-italian"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        Italian Bilingual
      </Styled(Text)>
      <Styled(Text)
        key="teach-spanish"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        Spanish Native
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Learn
      </Styled(Text)>
      <Styled(Text)
        key="learn-english"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        English Beginner A1/A2
      </Styled(Text)>
      <Styled(Text)
        key="learn-french"
        style={
          Object {
            "color": "#919191",
            "fontSize": 24,
          }
        }
      >
        French Fluent C1/C2
      </Styled(Text)>
      <Styled(Text)
        key="learn-german"
        style={
          Object {
            "color": "#919191",
            "fontSize": 24,
          }
        }
      >
        German Intermediate B1/B2
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`SearchScreen 2`] = `
<Styled(ViewNB)>
  <Loading />
  <Component />
</Styled(ViewNB)>
`;

exports[`SearchScreen 3`] = `
<Styled(ViewNB)>
  <Styled(Text)
    style={
      Object {
        "color": "#d9534f",
      }
    }
  >
    Impossible to load profiles
  </Styled(Text)>
  <Styled(ViewNB)>
    <Styled(Text)
      style={
        Object {
          "fontStyle": "italic",
          "textAlign": "center",
        }
      }
    >
      No user match your profile
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
